### ISSUE TITLE
Bug in function mg_close_connection when running WS server, no ssl?

### ISSUE THREAD

**Issue by hkhauke** [civetweb/civetweb#1066]
When using mg_close_connection on an active websocket server connection, the thread will not really end. The code is in civetweb.c, line 17804. Only setting the flag

`conn->must_close = 1;`

will not re-activate the listener thread which presumably hangs around in a recv or select.

I have modified the code at this position as follows:

```
if defined(USE_WEBSOCKET)
	if (conn->phys_ctx->context_type == CONTEXT_SERVER) {
		if (conn->in_websocket_handling) {
			/* Set close flag, so the server thread can exit. */
			conn->must_close = 1;

			// This is a fix to close a websocket server connection from within the server
			/*			
			return;
			*/
		}
	}
	else if (conn->phys_ctx->context_type == CONTEXT_WS_CLIENT) {

		unsigned int i;
```

This will then fall through to the `close_connection(conn);` rather than returning immediately.

Or do I use the mg_close_connection in a wrong way?

Thank you!

**Comment by bel2125** [comment]
Yes, Websocket server are in a receive loop that can be in another thread, and you only set a flag to exit that loop.

**Comment by hkhauke** [comment]
Thank you very much for the clarification,
